parser/src/lib.rs

```diff
@@ -24,33 +25,67 @@ pub mod token;
 mod token_reader;
 
 pub async fn parse(text: &str) -> Result<Ast, Error> {
-    let (lexer_in, lexer_out): (Sender<Token>, Receiver<Token>) = mpsc::channel(32);
-    let (normalizer_in, normalizer_out): (Sender<Token>, Receiver<Token>) = mpsc::channel(32);
+    let span = trace_span!("parsing");
+    let _enter = span.enter();
+    debug!(text);
+    let channel = 32;
+    trace!("reading channel {channel:?}");
+    let (lexer_in, lexer_out): (Sender<Token>, Receiver<Token>) =
+        mpsc::channel(channel);
+    debug!("successfully connected to lexer");
+    let (normalizer_in, normalizer_out): (Sender<Token>, Receiver<Token>) =
+        mpsc::channel(channel);
+    debug!("successfully connected to normalizer");
```

Comment:
No connection happens here, the channels are just created. So printing that things were successfully connected seems a bit misleading.